### ISSUE TITLE
Problem using js-phpize-phug with Phug

### ISSUE THREAD

**Issue by FerroLx** [pug-php/js-phpize-phug#7]
I'm installing Phug to use a template I bought that is written in pugjs. The problem is when i follow this:https://phug-lang.com/#use-javascript-expressions nothing happens.

I get the output:
userName

Could you help me?

or do you think is better to create the template pug file all over again in php?

Thanks in advance,
João Ferro

**Comment by kylekatarnls** [comment]
I use JS expressions very often with no problems, it's also the default behavior of [pug-php/pug](https://github.com/pug-php/pug)

If your pugjs template follows best practices and clean code style it will work as is with js-phpize-phug and Phug.

But I can't see what's wrong if you don't provide both the PHP code you use to configure Phug and render the template, and the template (at least the chunk that outputs `userName`).

**Comment by FerroLx** [comment]
I just installed the phug and still testing. the code I'm using is
<?php

/**
 * Created by PhpStorm.
 * User: joaoferro
 * Date: 2019-03-17
 * Time: 20:09
 */


use JsPhpize\JsPhpizePhug;


include_once __DIR__ . '/vendor/autoload.php';


Phug::setOption('debug', true);



Phug::display('p=userName', [
    'userName' => 'Bob',
]);



Phug::addExtension(JsPhpizePhug::class);


Phug::display('p=userName', [
    'userName' => 'Bob',
]);

**Comment by kylekatarnls** [comment]
I just ran `composer require phug/phug js-phpize/js-phpize-phug` in an empty directory to install Phug with js-phpize-phug, then create **index.php**:

```php
<?php

use JsPhpize\JsPhpizePhug;

include_once __DIR__ . '/vendor/autoload.php';

Phug::addExtension(JsPhpizePhug::class);

Phug::display('p=userName', [
  'userName' => 'Bob',
]);
```

Then I executed index.php and got:
```html
<p>Bob</p>
```

Please ensure you have both phug/phug and js-phpize/js-phpize-phug installed.

**Comment by FerroLx** [comment]
Thank you very much that worked. Now I'm having another problem 

Warning: Uncaught ErrorException: Object of class Phug\Lexer\Token\InterpolationStartToken could not be converted to string in /Users/joanamarote/PhpstormProjects/DevelopmentUFO/vendor/phug/lexer/src/Phug/Lexer/Analyzer/LineAnalyzer.php:178 Stack trace: #0 [internal function]: Phug\Util\SandBox->Phug\Util\{closure}(4096, 'Object of class...', '/Users/joanamar...', 178, Array) #1 /Users/joanamarote/PhpstormProjects/DevelopmentUFO/vendor/phug/lexer/src/Phug/Lexer/Analyzer/LineAnalyzer.php(178): implode('', Array) #2 [internal function]: Phug\Lexer\Analyzer\LineAnalyzer->Phug\Lexer\Analyzer\{closure}(Array) #3 /Users/joanamarote/PhpstormProjects/DevelopmentUFO/vendor/phug/lexer/src/Phug/Lexer/Analyzer/LineAnalyzer.php(179): array_map(Object(Closure), Array) #4 /Users/joanamarote/PhpstormProjects/DevelopmentUFO/vendor/phug/lexer/src/Phug/Lexer/Scanner/CommentScanner.php(45): Phug\Lexer\Analyzer\LineAnalyzer->getFlatLines() #5 /Users/joanamarote/PhpstormProjects/DevelopmentUFO/vendor/phug/lexer/src/Phug/Lexer/State.php(224 in /Users/joanamarote/PhpstormProjects/DevelopmentUFO/vendor/phug/lexer/src/Phug/Lexer/Analyzer/LineAnalyzer.php on line 178

Fatal error: Object of class Phug\Lexer\Token\ExpressionToken could not be converted to string in /Users/joanamarote/PhpstormProjects/DevelopmentUFO/vendor/phug/lexer/src/Phug/Lexer/Analyzer/LineAnalyzer.php on line 178

Now it's running the template in js but gives this error. I cannot understand what it is or which file is originating it. 

Sorry for bothering you!

Thanks,
João Ferro

**Comment by kylekatarnls** [comment]
The line 178 of **Lexer/Analyzer/LineAnalyzer.php** in the last version does not math any string conversion.

Please first update with `composer update`, re-run the code to get the error stack trace updated (if it's still in error) then give me your lexer version with:
`composer show phug/lexer`

**Comment by FerroLx** [comment]
Still in error:
`name     : phug/lexer
descrip. : Pug (ex-Jade) lexer for PHP, HTML template engine structured by indentation
keywords : dialect, html, jade, lexer, php, phtml, phug, presentation, pug, render, template, views
versions : * 0.5.33
type     : library
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
source   : [git] https://github.com/phug-php/lexer.git ef7e62075256503af62a42076b040ae82b5bc3ec
dist     : [zip] https://api.github.com/repos/phug-php/lexer/zipball/ef7e62075256503af62a42076b040ae82b5bc3ec ef7e62075256503af62a42076b040ae82b5bc3ec
path     : /Users/joanamarote/PhpstormProjects/DevelopmentUFO/vendor/phug/lexer
names    : phug/lexer

autoload
psr-4
* => ./src

requires
php >=5.5.0
phug/reader ^0.2.0
phug/util ^0.4.0

requires (dev)
kylekatarnls/multi-tester ^1.1
phug/dev-tool ^0.1.0`

**Comment by kylekatarnls** [comment]
Thanks, it must come from an indentation failure. I cannot guess the code that can causes the error, but I will add an error handler for this, so it could throw an clean error with the line number of the pug file where it happened.

**Comment by kylekatarnls** [comment]
Please try to install `composer require phug/lexer:dev-token-to-string` and re-run the code, it should give you a better message `Unexpected InterpolationStartToken inside raw text.` with the file and error line, so you could give me an example of what kind of code the lexer is failing on.

PS: if you have  a `minimum-stability` you will need to temporary remove it to install this version of the lexer.

Thanks for your patience.

**Comment by FerroLx** [comment]
Sorry for my question but I'm not very use to composer. How can I remove the minimum-stability?

I tried to put "minimum-stability": "dev", in my composer.json but didin't work.

Thanks

**Comment by kylekatarnls** [comment]
"minimum-stability": "dev" or no minimum-stability at all in your composer.json would be fine. Then you can run `composer require phug/lexer:dev-token-to-string`.

Maybe there is some cache so you may first need to run `composer clearcache`

**Comment by kylekatarnls** [comment]
Until I discover why there is this `InterpolationStartToken` (it's the token for `#{` inside texts) in a place the text should still be raw, I will let the error and publish a version with it. So it will ease your debug, just clear cache and update (no need to change minimum-stability):

```shell
composer clearcache
composer require phug/lexer:0.5.34
```

If it gives you a relevant line please give me the code chunk around and I will see if it's something we should support.

**Comment by kylekatarnls** [comment]
Hi, I consider this as fixed by the lexer 0.5.34. If you have an other issue, fill free to open a new one.
